Rebus.AdoNet/AdoNetTimeoutStorage.cs

```diff
@@ -105,20 +148,33 @@ public DueTimeoutsResult GetDueTimeouts()
 				{
 					while (reader.Read())
 					{
-						var sqlTimeout = DueAdoNetTimeout.Create(MarkAsProcessed, timeoutsTableName, reader);
+						var sqlTimeout = DueAdoNetTimeout.Create(MarkAsProcessed, timeoutsTableName, reader, connection, transaction);
 
 						dueTimeouts.Add(sqlTimeout);
-					}
+					}
 				}
 			}
 
-			return new DueTimeoutsResult(dueTimeouts);
+			return new DueTimeoutsResult(dueTimeouts, () => CommitAndClose(connection, transaction));
+		}
+
+		void CommitAndClose(IDbConnection connection, IDbTransaction transaction)
+		{
+			try
+			{
+				transaction.Commit();
+				transaction.Dispose();
+				connection.Dispose();
+			}
+			catch (Exception ex)
+			{
+				log.Error("CloseAndCommit produced an exception: {0}", ex.Message);
```

Comment:
When logging errors/exceptions, the whole exception object should be logged/passed to logging call, not just the message. (Otherwise the callstack and other information will be lost)